File: Back/src/main/java/com/showbook/back/entity/ShookLike.java
package com.showbook.back.entity;

import jakarta.persistence.Column;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.MapsId;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
public class ShookLike {
	@EmbeddedId
	private ShookLikePK shookLikePK;

	@MapsId(value = "shookId")
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "member_id", referencedColumnName = "member_id")
	private Member member;

	@MapsId(value = "shookId")
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "shook_id", referencedColumnName = "shook_id")
	private Shook shook;
	@Column(nullable = false)
	private Boolean likeStatus = false;
}

File: ml/flask/models/ShookLike.py
from config.dbConfig import db

class ShookLike(db.Model) :
    '''
    shook_id : shook_id
    member_id : 슈욱 좋아요 누른 member id
    like_status : 슈욱 좋아요 여부
    '''
    
    __tablename__ = "shook_like"
    
    __table_args__ = (db.PrimaryKeyConstraint('shook_id', 'member_id', name = 'shook_member_uc'), )
    
    shook_id = db.Column(db.Integer,db.ForeignKey("shook.shook_id"),nullable = False)
    member_id = db.Column(db.Integer,db.ForeignKey("member.member_id"),nullable = False)
    like_status = db.Column(db.Integer, nullable = False)
